uast/types.go

```diff
@@ -384,6 +385,26 @@ func setAnyOrNode(dst reflect.Value, n nodes.External) (bool, error) {
 		}
 		dst.Set(reflect.ValueOf(nd))
 		return true, nil
+	} else if rt == reflAnyArr {
+		narr, ok := n.(nodes.ExternalArray)
+		if !ok {
+			return false, nil
+		}
+		sz := narr.Size()
+		arr := make([]Any, 0, sz)
```

Comment:
Since you know the exact size anyway, and there's no early exit, you might as well just allocate and assign rather than appending: `make([]Any, sz)` … `arr[i] = v`

Reply:
I usually prefer not to. Because an exact size is only an optimization, and shouldn't affect the code below.

Reply:
Generally if you don't care about optimality, I recommend the simpler code: `var arr []Any`. The `append` builtin reallocates intelligently to avoid quadratic moves. If the allocation really matters, it's better not to append at all, and assign directly. The only time this pattern really makes sense is if you're copying out of a map, where there is no index without a separate explicit induction variable.

Reply:
It's more like a middle ground. When I don't care about optimality, I use `var arr []Any`, but when I do care, I simply change that single line to `arr := make([]Any, 0, sz)` and don't touch the code below (it's valid for both cases).

You may argue that it's might be less optimal than a direct assignment, since `append` will increment the counter, but it should be negligible comparing to any allocation and the code is a bit more flexible in case some decide to add additional items. I think it's a good balance, but maybe I should use the optimal one, as you suggest.